binlog.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of the MAVLink Router project
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "binlog.h"
+
+#include <assert.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "log.h"
+#include "mainloop.h"
+#include "util.h"
+
+bool BinLog::_start_timeout()
+{
+    mavlink_message_t msg;
+
+    mavlink_msg_remote_log_block_status_pack(_system_id, MAV_COMP_ID_ALL, &msg, _target_system_id, MAV_COMP_ID_ALL, MAV_REMOTE_LOG_DATA_BLOCK_START, 1);
+    _send_msg(&msg, _target_system_id);
+
+    return true;
+}
+
+bool BinLog::start()
+{
+    if (!LogEndpoint::start()) {
+        return false;
+    }
+
+    _last_acked_seqno = 0;
+    _system_id = SYSTEM_ID;
+
+    return true;
+}
+
+void BinLog::stop()
+{
+    mavlink_message_t msg;
+
+    if (_file == -1) {
+        log_error("BinLog not started");
+        return;
+    }
+
+    mavlink_msg_remote_log_block_status_pack(_system_id, MAV_COMP_ID_ALL, &msg, _target_system_id, MAV_COMP_ID_ALL, MAV_REMOTE_LOG_DATA_BLOCK_STOP, 1);
+    _send_msg(&msg, _target_system_id);
+
+    LogEndpoint::stop();
+}
+
+int BinLog::write_msg(const struct buffer *buffer)
+{
+    const bool mavlink2 = buffer->data[0] == MAVLINK_STX;
+    uint32_t msg_id;
+    uint16_t payload_len;
+    uint8_t trimmed_zeros;
+
+    if (mavlink2) {
+        struct mavlink_router_mavlink2_header *msg
+            = (struct mavlink_router_mavlink2_header *)buffer->data;
+        msg_id = msg->msgid;
+        payload_len = msg->payload_len;
+    } else {
+        struct mavlink_router_mavlink1_header *msg
+            = (struct mavlink_router_mavlink1_header *)buffer->data;
+        msg_id = msg->msgid;
+        payload_len = msg->payload_len;
+    }
+
+    /* Check if we are interested in this msg_id */
+    if (msg_id != MAVLINK_MSG_ID_REMOTE_LOG_DATA_BLOCK) {
+        return buffer->len;
+    }
+
+    _stat.write.total++;
+    _stat.write.bytes += buffer->len;
+
+    if (_logging_start_timeout) {
+        _remove_start_timeout();
```

Comment:
@edersondisouza Sorry, yes, you should be dropping them.  Just have to rely on the server to get the idea that noone's listening.